gateway-js/src/datasources/RemoteGraphQLDataSource.ts

```diff
@@ -225,6 +181,58 @@ export class RemoteGraphQLDataSource<
     }
   }
 
+  async sendRequest(
+    options: GraphQLDataSourceProcessOptions
+  ): Promise<{ request: GatewayGraphQLRequest; response: GatewayGraphQLResponse }> {
+    const { request, context } = options;
+
+    if (!request.query) {
+      throw new Error('Missing query');
+    }
+
+    const { query, ...requestWithoutQuery } = request;
+
+    // If APQ was enabled, we'll run the same request again, but add in the
+    // previously omitted `query`.  If APQ was NOT enabled, this is the first
+    // request (non-APQ, all the way).
+    if (this.apq) {
+      const apqHash = createHash('sha256').update(request.query).digest('hex');
+
+      // Take the original extensions and extend them with
+      // the necessary "extensions" for APQ handshaking.
+      requestWithoutQuery.extensions = {
+        ...request.extensions,
+        persistedQuery: {
+          version: 1,
+          sha256Hash: apqHash,
+        },
+      };
+
+      const apqOptimisticResponse = await this._sendRequest(requestWithoutQuery, context as TContext);
```

Comment:
had to cast `context` here as well as is done in `process`, could probably be refactored to only do it once somewhere else